crates/neon-runtime/src/neon_task.h

```diff
@@ -0,0 +1,93 @@
+#ifndef NEON_TASK_H_
+#define NEON_TASK_H_
+
+#include <nan.h>
+#include <uv.h>
+#include "neon.h"
+#include "v8.h"
+
+namespace neon {
+
+class Task {
+public:
+  Task(v8::Isolate *isolate,
+       void *rust_task,
+       Neon_TaskPerformCallback perform,
+       Neon_TaskCompleteCallback complete,
+       v8::Local<v8::Function> callback)
+    : isolate_(isolate),
+      rust_task_(rust_task),
+      perform_(perform),
+      complete_(complete)
+  {
+    request_.data = this;
+    result_ = nullptr;
+    callback_.Reset(isolate, callback);
+  }
+
+  ~Task() {
+    callback_.Reset();
+  }
+
+  void execute() {
+    result_ = perform_(rust_task_);
+  }
+
+  void complete() {
+    Nan::HandleScope scope;
```

Comment:
You should enter an Isolate::Scope and Context::Scope here:
```c++
v8::Isolate::Scope isolate_scope(isolate_);
v8::HandleScope handle_scope(isolate_);  // Or Nan::HandleScope
v8::Local<v8::Context> context = v8::Local<v8::Context>::New(isolate_, context_);
v8::Context::Scope context_scope(context);
// ...
```
Store the context when the task is created because it can change.

Using `v8::Isolate::GetCurrentContext()` in this method is not sound because it's possible no context has been entered or it's a different one from the one that created the task.